skfp/distances/all_bit.py

```diff
@@ -0,0 +1,365 @@
+from typing import Union
+
+import numpy as np
+from numba import njit, prange
+from scipy.sparse import csr_array
+from sklearn.utils._param_validation import validate_params
+
+from .utils import _check_nan
+
+
+@validate_params(
+    {
+        "vec_a": ["array-like", csr_array],
+        "vec_b": ["array-like", csr_array],
+    },
+    prefer_skip_nested_validation=True,
+)
+def all_bit_binary_similarity(
+    vec_a: Union[np.ndarray, csr_array], vec_b: Union[np.ndarray, csr_array]
+) -> float:
+    r"""
+    Calculate the all-bit binary similarity between two binary vectors.
+
+    Computes the all-bit similarity for binary data between two input arrays
+    or sparse matrices using the formula:
+
+    .. math::
+
+        sim(vec_a, vec_b) = \frac{\text{count}_a - \text{count}_\text{xor}}{\text{count}_a}
+
+
+    where:
+
+    - :math:`\text{count}_a` is the number of non-zero elements in :math:`\mathbf{a}`,
+    - :math:`\text{count}_\text{xor}` is the number of positions where the binary values in :math:`\mathbf{a}` and :math:`\mathbf{b}` differ
+
+    The calculated similarity falls within the range ``[0, 1]``.
+    Passing all-zero vectors to this function results in a similarity of 1.
```

Comment:
This similarity is an exception to this. All-zero vectors have no bits in common, so they have similarity 0. This is common for all metrics using only "on" bits, see e.g. https://pmc.ncbi.nlm.nih.gov/articles/PMC8026075/